### ISSUE TITLE
Can't get back to hammerspoon preferences

### ISSUE THREAD

**Issue by wkf** [Hammerspoon/hammerspoon#1475]
Hello there, 

I disabled both the dock and the menu bar icons. However, when I launch hammerspoon from spotlight or the terminal (`open -a Hammerspoon`), I'm greeted with only the console, not the preference window. Am I missing something?

Thanks!

**Comment by cmsj** [comment]
fwiw you can open the preferences window from the console (or bound to a hotkey or whatever other hammerspoon config you choose) with http://www.hammerspoon.org/docs/hs.html#openPreferences

I think we felt that overall, having the console appear on relaunch, was probably more useful than the prefs pane, since there are very few prefs.

**Comment by Dbz** [comment]
I believe it is important to always be able to intuitively open preferences. Hammerspoon now requires users who made a mistake to google around until they figure out a solution. Please consider reopening this issue.

**Comment by cmsj** [comment]
@Dbz FWIW, the normal Mac shortcut for summoning prefs windows, ⌘-, works from the Console window.

Also FWIW, I'm generally not supportive of Hammerspoon being a dependency of things. This is a complicated, powerful tool that is intended for programmers to use and we don't have the human bandwidth to deal with support issues from folk it's not aimed at. I know that sounds harsh, but this was always intended to be a nuts-and-bolts toolkit for people hitting the limits of the various excellent, but not programmable, Mac automation apps.

Having said all that, I wouldn't be opposed to a toolbar in the console window which included a button to open the prefs window. @asmagill - what do you think about having a `hs.console.defaultToolbar` variable with some number of useful buttons in it?

**Comment by asmagill** [comment]
The `hs.webview.toolbar` module allows you to create a toolbar and add it to a webview or the console already.  See the docs for `hs.webview.toolbar` and `hs.console.toolbar`.

**Comment by cmsj** [comment]
@asmagill yep, that's what I was thinking we'd use, but I was wondering about having the default toolbar available as a variable (or maybe better, a constant).

**Comment by asmagill** [comment]
Ah, missed that in my skimming... wouldn't be hard, the question is what to put into it and how flexible do we make it? Do we make it just a set of presets (preferences and what else?) or do we make a wrapper with methods so that the user can add/remove an image, title and function to it without having to create their own toolbar from scratch with the `hs.webview.toolbar` module?

I'm not sure I agree that Hammerspoon makes it unintuitive to get to the preferences, though... the standard Mac Cmd-Comma works when it's in the foreground, it's in the menubar under the Application menu if you haven't hidden the dock icon, and its in the built in menu if you haven't disabled the "show menu icon" preference... as well as the console with a command in the root documentation tree (i.e. `hs`)... where else would we put a way to get to it?

**Comment by cmsj** [comment]
@asmagill I would just make it a static set of presets (and tbh Prefs is the only thing I can think of that would go there right now. Maybe Reload/Quit?). I don't see a need to make it be particularly flexible since people can set their own with `hs.console.toolbar()`.

FWIW this is for people who have disabled the dock and the menu icon.

**Comment by asmagill** [comment]
Add `hs.doc.hsdocs.help()` in there as well to pop up the built in doc viewer.

**Comment by cmsj** [comment]
That's actually a really great idea :)

**Comment by asmagill** [comment]
Ok, actually yeah, it makes a little more sense to me with that... it's been a while since I played with the toolbar (I've upgraded machines since then and have been slowly moving pieces of my old config over to the new machine so I can see what's slow and might need tweaking and what I actually use... if you look at https://github.com/asmagill/hammerspoon-config/blob/master/utils/consoleToolbar.lua (which I warn you is poorly documented and does more then we're talking about) I also use the search field in the toolbar so that I could type in a module and have it pop up the doc viewer right at that specific module, plus use the pop-up menu feature of the search field to have a predefined list of module names as well...

Maybe more complex then we need as a first go, but something to keep in mind. It can always be added later.

**Comment by latenitefilms** [comment]
Having a default toolbar is a cool idea!

I also wonder if the original issue could also be addressed by adding a "Preferences" option (and "Console" for that matter) to the Dock Icon (issue #1350)?

Personally though, going back to the original issue, if the user is advanced enough to know how to code things to disable both the dock and the menu bar icons, then you'd ASSUME that they're advanced enough to enable them again?

**Comment by cmsj** [comment]
I've just merged #1625 so this should now be covered :)

**Comment by brad** [comment]
TL;DR If you are stuck, enter `hs.openPreferences()` into the console
